src/lib.rs

```diff
@@ -175,24 +216,11 @@ fn into_forward_req(
 }
 
 #[instrument]
-async fn forward_request(req: Request<Incoming>) -> Result<Response<Incoming>, Error> {
-    #[cfg(not(feature = "_test-util"))]
-    let builder = HttpsConnectorBuilder::new().with_webpki_roots();
-
-    // During testing we require self signed certificates, so if SSL_CERT_FILE
-    // is explicitly set under such builds, parse root certificates from the
-    // specified PEM file
-    #[cfg(feature = "_test-util")]
-    let builder = if std::env::var("SSL_CERT_FILE").is_ok() {
-        HttpsConnectorBuilder::new()
-            .with_native_roots()
-            .expect("SSL_CERT_FILE provided certificates must load")
-    } else {
-        HttpsConnectorBuilder::new().with_webpki_roots()
-    };
-
-    let https = builder.https_or_http().enable_http1().build();
-    let client = Client::builder(TokioExecutor::new()).build(https);
+async fn forward_request(
+    req: Request<Incoming>,
+    config: &RelayConfig,
+) -> Result<Response<Incoming>, Error> {
+    let client = config.client.clone();
     client.request(req).await.map_err(|_| Error::BadGateway)
```

Comment:
I don't even think you need to `.0.clone()`, do you?

``sugg
```suggestion
    config.client.request(req).await.map_err(|_| Error::BadGateway)
```